File: server/kafka-backend/src/core/domain/events/transfer-failed.event.ts
export class TransferFailedEvent
{
  constructor(
    public readonly trackingId: string,
    public readonly fromAccountId: string,
    public readonly toAccountId: string,
    public readonly fromIban: string,
    public readonly toIban: string,
    public readonly amount: number,
    public readonly currency: string,
    public readonly failureReason: string,
    public readonly failedAt: Date = new Date(),
  ) {}

  static eventName = 'transfer.failed';
}

File: server/java-backend/src/main/java/com/intbank/core/domain/event/TransferFailedEvent.java
package com.intbank.core.domain.event;

import java.time.Instant;

public record TransferFailedEvent(
    String trackingId,
    String fromAccountId,
    String toAccountId,
    String fromIban,
    String toIban,
    double amount,
    String currency,
    String failureReason,
    Instant failedAt
)
{
    public static final String EVENT_NAME = "transfer.failed";
}